.github/actions/setup_env/action.yml

```diff
@@ -0,0 +1,42 @@
+name: "Set up python and conda environment"
+inputs:
+  cache-id:
+    description:
+      "Cache identifier used in the cache key. This should include any values
+      needed to uniquely identify the appropriate cache, such as python
+      version and operating system. The number of the current week in the
+      year, and a hash of all requirements files, are appended onto this
+      identifier to make the cache key. This way the cache will refresh each
+      week, or if the requirements change."
+    required: false
+    default: ""
+outputs:
+  cache-hit:
+    description: "True if a cache hit was found, false if not"
+    value: ${{ steps.condacache.outputs.cache-hit }}
+runs:
+  using: "composite"
```

Comment:
What's `composite`?

Reply:
`composite` is the type of the github action. It means it combines multiple steps that could ordinarily be run directly in a workflow. As opposed to Docker or Javascript actions, which don't have a `steps` section - they point to a Dockerfile or javascript file that contains the action logic.

A `composite` action works here because it's just combining other actions and some shell commands. To do something more complicated, you'd need to use docker or javascript.